fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationManager.java

```diff
@@ -113,40 +117,92 @@ public UserIdentity checkPassword(String remoteUser, String remoteHost, byte[] r
     }
 
     public void createUser(CreateUserStmt stmt) throws DdlException {
+        UserIdentity userIdentity = stmt.getUserIdent();
+        UserAuthenticationInfo info = stmt.getAuthenticationInfo();
+        writeLock();
         try {
-            // validate authentication info by plugin
-            String pluginName = stmt.getAuthPlugin();
-            AuthenticationProvider provider = AuthenticationProviderFactory.create(pluginName);
-            UserIdentity userIdentity = stmt.getUserIdent();
-            UserAuthenticationInfo info = provider.validAuthenticationInfo(
-                    userIdentity, stmt.getOriginalPassword(), stmt.getAuthString());
-            info.setAuthPlugin(pluginName);
-            info.setOrigUserHost(userIdentity.getQualifiedUser(), userIdentity.getHost());
-
-            writeLock();
-            try {
-                updateUserNoLock(userIdentity, info, false);
-                UserProperty userProperty = null;
-                if (!userNameToProperty.containsKey(userIdentity.getQualifiedUser())) {
-                    userProperty = new UserProperty();
-                    userNameToProperty.put(userIdentity.getQualifiedUser(), userProperty);
-                }
-                GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
-                PrivilegeManager privilegeManager = globalStateMgr.getPrivilegeManager();
-                // init user privilege
-                UserPrivilegeCollection collection = privilegeManager.onCreateUser(userIdentity);
-                short pluginId = privilegeManager.getProviderPluginId();
-                short pluginVersion = privilegeManager.getProviderPluginVerson();
-                globalStateMgr.getEditLog().logCreateUser(
-                        userIdentity, info, userProperty, collection, pluginId, pluginVersion);
-            } finally {
-                writeUnlock();
+            updateUserNoLock(userIdentity, info, false);
+            UserProperty userProperty = null;
+            if (!userNameToProperty.containsKey(userIdentity.getQualifiedUser())) {
```

Comment:
For legacy reason, userNameToProperty used username instead of username + host as the key. In other words,`test`@`%` and `test`@`10.1.1` share the same properties.

Reply:
got it